build/build-bundle.js

```diff
@@ -49,155 +36,177 @@ const isLightrider = file => path.basename(file).includes('lightrider');
-  await new Promise((resolve, reject) => {
-    const writeStream = fs.createWriteStream(distPath);
-    writeStream.on('finish', resolve);
-    writeStream.on('error', reject);
+  /** @type {Record<string, string>} */
+  const shimsObj = {};
 
-    // Extract the inline source map to an external file.
-    if (DEBUG) bundleStream = bundleStream.pipe(exorcist(`${distPath}.map`));
-    bundleStream.pipe(writeStream);
-  });
+  const modulesToIgnore = [
+    'intl-pluralrules',
+    'intl',
+    'pako/lib/zlib/inflate.js',
+    'raven',
+    'source-map',
+    'ws',
+    require.resolve('../lighthouse-core/gather/connections/cri.js'),
+  ];
 
-  if (isDevtools(distPath)) {
-    let code = fs.readFileSync(distPath, 'utf-8');
-    // Add a comment for TypeScript, but not if in DEBUG mode so that source maps are not affected.
-    // See lighthouse-cli/test/smokehouse/lighthouse-runners/bundle.js
-    if (!DEBUG) {
-      code = '// @ts-nocheck - Prevent tsc stepping into any required bundles.\n' + code;
-    }
+  // Don't include the stringified report in DevTools - see devtools-report-assets.js
+  // Don't include in Lightrider - HTML generation isn't supported, so report assets aren't needed.
+  if (isDevtools(entryPath) || isLightrider(entryPath)) {
+    modulesToIgnore.push(require.resolve('../report/generator/report-assets.js'));
+  }
 
-    // DevTools build system expects `globalThis` rather than setting global variables.
-    assert.ok(code.includes('\nrequire='), 'missing browserify require stub');
-    code = code.replace('\nrequire=', '\nglobalThis.require=');
-    assert.ok(!code.includes('\nrequire='), 'contained unexpected browserify require stub');
+  // Don't include locales in DevTools.
+  if (isDevtools(entryPath)) {
+    const localeKeys = Object.keys(require('../lighthouse-core/lib/i18n/locales.js'));
+    /** @type {Record<string, {}>} */
+    const localesShim = {};
+    for (const key of localeKeys) localesShim[key] = {};
+    shimsObj['./locales.js'] = `export default ${JSON.stringify(localesShim)}`;
+  }
 
-    fs.writeFileSync(distPath, code);
+  for (const modulePath of modulesToIgnore) {
+    shimsObj[modulePath] = 'export default {}';
   }
-}
 
-/**
- * Minify a javascript file, in place.
- * @param {string} filePath
- */
-async function minifyScript(filePath) {
-  const code = fs.readFileSync(filePath, 'utf-8');
-  const result = await terser.minify(code, {
-    ecma: 2019,
-    output: {
-      comments: /^!|Prevent tsc/,
-      max_line_len: 1000,
-    },
-    // The config relies on class names for gatherers.
-    keep_classnames: true,
-    // Runtime.evaluate errors if function names are elided.
-    keep_fnames: true,
-    sourceMap: DEBUG && {
-      content: JSON.parse(fs.readFileSync(`${filePath}.map`, 'utf-8')),
-      url: path.basename(`${filePath}.map`),
-    },
+  shimsObj[require.resolve('../package.json')] =
+    `export const version = ${JSON.stringify(require('../package.json').version)}`;
```

Comment:
It's wrapped in stringify because it emits valid JS.

`version = ${pkg.version)}` results in `version = 8.5.1`.

one could argue for `version = '${pkg.version}'` but the quotes are a little subtle. also, not relevant here but that pattern breaks when there are invalid string characters (this was a big motivator for the `ExecutionContext.evaluate` api).